prototype_source/backend_config_tutorial.rst

```diff
@@ -0,0 +1,410 @@
+(prototype) PyTorch BackendConfig Tutorial
+==============================
+**Author**: `Andrew Or <https://github.com/andrewor14>`_
+
+The BackendConfig API enables developers to integrate their backends
+with PyTorch quantization. It is currently only supported in FX graph
+mode quantization, but support may be extended to other modes of
+quantization in the future. In this tutorial, we will demonstrate how to
+use this API to customize quantization support for specific backends.
+For more information on the motivation and implementation details behind
+BackendConfig, please refer to this
+`README <https://github.com/pytorch/pytorch/tree/master/torch/ao/quantization/backend_config>`__.
+
+BackendConfig API Specification
+-------------------------------
+
+On a high level, BackendConfig specifies the quantization behavior for
+each supported operator pattern (e.g. linear, conv-bn-relu). The API is
+broken down into the following class hierarchy:
+
+- `BackendConfig <https://pytorch.org/docs/stable/generated/torch.ao.quantization.backend_config.BackendConfig.html>`__:
+  The main class to be passed to prepare and convert functions
+- `BackendPatternConfig <https://pytorch.org/docs/stable/generated/torch.ao.quantization.backend_config.BackendPatternConfig.html>`__:
+  Config object that specifies quantization behavior for a given
+  operator pattern. Each BackendConfig consists of many of these.
+- `DTypeConfig <https://pytorch.org/docs/stable/generated/torch.ao.quantization.backend_config.DTypeConfig.html>`__:
+  Config object that specifies the supported data types and constraints
+  (if any) for input and output activations, weights, and biases. Each
+  BackendPatternConfig consists of one or more of these.
+- `DTypeWithConstraints <https://pytorch.org/docs/stable/generated/torch.ao.quantization.backend_config.DTypeWithConstraints.html>`__:
+  Constraints imposed by the backend on the quantization parameters
+  (scale and zero point) and ranges when quantizing to a given data
+  type. Each DTypeConfig consists of many of these.
+
+Pattern Specification
```

Comment:
I remember that we decided a few months ago we cannot ship the current pattern format to users as it's confusing.  Do we have any thoughts on when we plan to switch to a more intuitive format?

Reply:
Yes, we had a design review on this but the implementation was deprioritized due to other projects. We should revisit the priority of this next half since changing the frontend should be relatively low effort

Reply:
I think it makes sense for fixes like these to block user facing documents.  We really shouldn't be shipping stuff like this to OSS users.